src/main/java/org/folio/entlinks/service/links/AuthorityDataStatService.java

```diff
@@ -50,7 +50,15 @@ public List<AuthorityDataStat> createInBatch(List<AuthorityDataStat> stats) {
 
   public List<AuthorityDataStat> fetchDataStats(OffsetDateTime fromDate, OffsetDateTime toDate,
                                                 AuthorityDataStatActionDto action, int limit) {
-    return statRepository.findByDateAndAction(AuthorityDataStatAction.valueOf(action.getValue()),
+    AuthorityDataStatAction authorityDataStatAction = switch (action) {
+      case DELETE -> AuthorityDataStatAction.DELETE;
+      case UPDATE_NATURAL_ID -> AuthorityDataStatAction.UPDATE_NATURAL_ID;
+      case UPDATE_HEADING -> AuthorityDataStatAction.UPDATE_HEADING;
+      default ->
+        throw new IllegalArgumentException("Unexpected enum constant: " + action);
+    };
+
+    return statRepository.findByDateAndAction(authorityDataStatAction.name(),
```

Comment:
Why this change is needed? I believe the previous code version should handle this logic.